collector/src/output_frontend.cpp

```diff
@@ -518,6 +568,74 @@ size_t CMonitorOutputFrontend::get_current_sample_measurements() const
     return ntotal_meas;
 }
 
+#ifdef PROMETHEUS_SUPPORT
+void CMonitorOutputFrontend::push_current_sections_to_prometheus()
+{
+    std::string metric_name;
+    std::map<std::string, std::string> lbl;
+    for (size_t i = 0; i < m_current_sections.size(); i++) {
+        auto& sec = m_current_sections[i];
+        metric_name = sec.m_name;
+        if (sec.m_measurements.empty()) {
+            for (size_t i = 0; i < sec.m_subsections.size(); i++) {
+                auto& subsec = sec.m_subsections[i];
+                if (subsec.m_measurements.empty()) {
+                    for (size_t i = 0; i < subsec.m_subsubsections.size(); i++) {
+                        auto& subsubsec = subsec.m_subsubsections[i];
+                        for (size_t n = 0; n < subsubsec.m_measurements.size(); n++) {
+                            auto& measurement = subsubsec.m_measurements[n];
+                            lbl = { { "metric", subsubsec.m_name } };
+                            if (!subsubsec.m_labels.empty()) {
```

Comment:
this "if" condition and the logic to insert into "lbl" all the sub-subsection labels can be moved "one level up" outside the for loop done on measurements. In this way if a sub-subsection contains 100 measurements, the code executes just once, right?